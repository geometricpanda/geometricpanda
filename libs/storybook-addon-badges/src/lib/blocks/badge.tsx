import React, { FC } from 'react';
import { styled } from '@storybook/theming';
import { WithBadgeTooltip } from './withBadgeTooltip';
import type { BadgeProps } from '../types';
import { getBadgeCustomProperty, getTitle } from '../helpers/helpers';

export const BadgeWrapper = styled.span(({ theme }) => ({
  display: 'flex',
  paddingInline: theme.layoutMargin / 2,
}));

export const BadgeInner = styled.span<Omit<BadgeProps, 'config'>>(
  ({ badge }) => ({
    borderColor: `var(${getBadgeCustomProperty(badge, 'border-color')})`,
    borderRadius: `var(${getBadgeCustomProperty(badge, 'border-radius')})`,
    borderStyle: `var(${getBadgeCustomProperty(badge, 'border-style')})`,
    borderWidth: `var(${getBadgeCustomProperty(badge, 'border-width')})`,
    color: `var(${getBadgeCustomProperty(badge, 'color')})`,
    backgroundColor: `var(${getBadgeCustomProperty(
      badge,
      'background-color'
    )})`,
    fontSize: `var(${getBadgeCustomProperty(badge, 'font-size')})`,
    fontFamily: `var(${getBadgeCustomProperty(badge, 'font-family')})`,
    'font-weight': `var(${getBadgeCustomProperty(badge, 'font-weight')})`,
    lineHeight: `var(${getBadgeCustomProperty(badge, 'line-height')})`,
    'text-transform': `var(${getBadgeCustomProperty(badge, 'text-transform')})`,
    paddingInline: `var(${getBadgeCustomProperty(badge, 'padding-inline')})`,
    paddingBlock: `var(${getBadgeCustomProperty(badge, 'padding-block')})`,
    display: 'block',
  })
);

const BadgeInnerWithTooltip = WithBadgeTooltip(BadgeInner);

export const Badge: FC<BadgeProps> = ({ badge, config }) => (
  <BadgeWrapper>
    <BadgeInnerWithTooltip badge={badge} config={config}>
      {getTitle(badge, config)}
    </BadgeInnerWithTooltip>
  </BadgeWrapper>
);
