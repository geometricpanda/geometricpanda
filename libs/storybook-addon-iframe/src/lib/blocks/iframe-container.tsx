import styled from 'styled-components';

export interface IFrameContainerProps {
  loaded?: boolean
}

export const IFrameContainer = styled.div<IFrameContainerProps>`
  display: flex;
  height: 100%;
  transition: transform 300ms;
  transform: ${({ loaded }) => loaded ? 'scale(1)' : 'scale(0)'};
`;
