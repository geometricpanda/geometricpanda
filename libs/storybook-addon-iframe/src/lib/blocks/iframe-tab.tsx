import { Group, Story } from '@storybook/api';
import React, { useLayoutEffect, useState } from 'react';
import { EmbeddedIframeLayout, GenericLayout } from '../layouts';
import { NotFoundIcon } from '../icons';

export interface OtherTabProps {
  parameters: Story['parameters'] | Group['parameters']
}

export const IframeTab: React.FC<OtherTabProps> = ({ parameters }) => {
  const [src, setSrc] = useState<string>(null);

  useLayoutEffect(() => {
    setSrc(parameters['iframeSrc'] || null);
  }, [parameters]);

  return src
    ? <EmbeddedIframeLayout src={src} />
    : <GenericLayout icon={<NotFoundIcon/>}>Nothing to see</GenericLayout>;
};
