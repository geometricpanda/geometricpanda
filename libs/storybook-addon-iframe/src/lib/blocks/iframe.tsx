import { styled } from '@storybook/theming'

export const IFrame = styled.iframe`
  background-color: white;
  border: 0;
  height: 100%;
  width: 100%;
`;
