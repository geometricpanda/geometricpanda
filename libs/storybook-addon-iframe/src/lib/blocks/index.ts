export * from './iframe';
export * from './iframe-container';
