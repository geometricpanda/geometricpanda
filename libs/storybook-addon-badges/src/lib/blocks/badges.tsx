import React, { FC } from 'react';
import { Separator, Spaced } from '@storybook/components';

import type { BadgesConfig } from '../types';

import { defaultBadgesConfig } from '../shared';
import { Badge } from './badge';

interface BadgesProps {
  badges?: string[];
  badgesConfig?: BadgesConfig;
}

export const Badges: FC<BadgesProps> = ({ badges = [], badgesConfig = defaultBadgesConfig }) => (
  <>
    <Separator />
    {badges.map((badge, index) =>
      <Spaced row={1} col={index} key={badge}>
        <Badge badge={badge} config={badgesConfig[badge]} />
      </Spaced>
    )}
    <Separator />
  </>
);
