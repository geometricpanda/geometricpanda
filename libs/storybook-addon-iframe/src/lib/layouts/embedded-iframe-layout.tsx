import React, { useCallback, useLayoutEffect, useRef, useState } from 'react';
import { IFrame } from '../blocks/iframe';
import { IFrameContainer } from '../blocks/iframe-container';
import { GenericLayout } from './generic-layout';
import { ErrorIcon, SpeedIcon } from '../icons';

export interface EmbeddedIframeProps {
  url: string;
  timeout: number;
}

export enum LOADING_STATE {
  INITIAL = 'initial',
  LOADING = 'loading',
  LOADED = 'loaded',
  TIMEOUT = 'timeout',
}

export const EmbeddedIframeLayout: React.FC<EmbeddedIframeProps> = ({ url, timeout }) => {

  const [state, setState] = useState<LOADING_STATE>(LOADING_STATE.INITIAL);
  const errorTimeout = useRef<NodeJS.Timeout>(null);

  const onUnmount = useCallback(() => {
    setState(LOADING_STATE.INITIAL);
    clearTimeout(errorTimeout.current);
  }, [setState, errorTimeout]);

  const onLoad = () => {
    setState(LOADING_STATE.LOADED);
    clearTimeout(errorTimeout.current);
  };

  const onTimeout = () => {
    setState(LOADING_STATE.TIMEOUT);
  };

  useLayoutEffect(() => {
    setState(LOADING_STATE.LOADING);
    errorTimeout.current = setTimeout(onTimeout, timeout);
    return onUnmount;
  }, [onUnmount, setState, errorTimeout, timeout]);

  switch (state) {
    case LOADING_STATE.LOADING:
    case LOADING_STATE.LOADED:
      return (
        <>
          {state === LOADING_STATE.LOADING && (
            <GenericLayout data-test-id={'loading'} icon={<SpeedIcon />}>
              Loading
            </GenericLayout>)
          }
          <IFrameContainer loaded={state === LOADING_STATE.LOADED}>
            <IFrame
              data-test-loaded={state === LOADING_STATE.LOADED}
              onLoad={onLoad}
              scrolling={'auto'}
              src={url}
            />
          </IFrameContainer>
        </>
      );

    case LOADING_STATE.TIMEOUT:
      return (
        <GenericLayout data-test-id={'error'} icon={<ErrorIcon />}>
          Sorry, this tab didn't load in time
        </GenericLayout>
      );

    case LOADING_STATE.INITIAL:
    default:
      return null;
  }

};
