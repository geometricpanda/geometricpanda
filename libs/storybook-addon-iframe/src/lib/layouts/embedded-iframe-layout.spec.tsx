import { EmbeddedIframeLayout } from './embedded-iframe-layout';

jest.mock('../blocks/iframe', () => ({
  IFrame: ({ onLoad }) => <div data-test-id={'iframe-mock'} />,
  default: null
}));

import renderer from 'react-test-renderer';
import React from 'react';
import { IFrame, IFrameContainer } from '../blocks';

describe('EmbeddedIframeLayout', () => {

  it('should show a loading block', () => {
    const tree = renderer.create(<EmbeddedIframeLayout src={'https://www.google.com'} />);
    const testInstance = tree.root;
    const loadingBlocks = testInstance.findAllByProps({ 'data-test-id': 'loading' });
    expect(loadingBlocks.length).toBe(1);
    expect(tree).toMatchSnapshot();
  });

  it('should show an error block if loading has not completed within 10 seconds', () => {
    jest.useFakeTimers();

    const tree = renderer.create(<EmbeddedIframeLayout src={'https://www.google.com'} />);
    const testInstance = tree.root;

    renderer.act(() => jest.advanceTimersByTime(10000));

    const errorBlocks = testInstance.findAllByProps({ 'data-test-id': 'error' });
    expect(errorBlocks.length).toBe(1);
    expect(tree).toMatchSnapshot();

    jest.useRealTimers();
  });

  it('should only show the iframe block on success', () => {
    jest.useFakeTimers();

    const tree = renderer.create(<EmbeddedIframeLayout src={'https://www.google.com'} />);
    const testInstance = tree.root;

    const iFrame = testInstance.findByType(IFrame);

    renderer.act(() => {
      iFrame.props.onLoad();
      jest.advanceTimersByTime(10000);
    });

    const errorBlocks = testInstance.findAllByProps({ 'data-test-id': 'error' });
    const loadingBlocks = testInstance.findAllByProps({ 'data-test-id': 'loading' });

    expect(errorBlocks.length).toEqual(0);
    expect(loadingBlocks.length).toEqual(0);
    expect(iFrame.props['data-test-loaded']).toBeTruthy();
    expect(tree).toMatchSnapshot();

    jest.useRealTimers();
  });


});
