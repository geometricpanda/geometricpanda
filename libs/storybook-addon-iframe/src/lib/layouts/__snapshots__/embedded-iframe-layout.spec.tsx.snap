// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedIframeLayout should only show the iframe block on success 1`] = `
<div
  className="sc-1iqrqt3-0 crXgSr"
>
  <div
    data-test-id="iframe-mock"
  />
</div>
`;

exports[`EmbeddedIframeLayout should show a loading block 1`] = `
Array [
  <div
    className="es7h93-2 fWpopI"
  >
    <div
      className="es7h93-0 fOAtDc"
    >
      <svg
        viewBox="1 -16 511.999 511"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m480.683594 472.882812h-449.367188c-13.152344 0-23.816406-10.664062-23.816406-23.816406v-417.25c0-13.152344 10.664062-23.816406 23.816406-23.816406h449.367188c13.15625 0 23.816406 10.664062 23.816406 23.816406v417.25c0 13.152344-10.664062 23.816406-23.816406 23.816406zm0 0"
          fill="#f4f4f4"
        />
        <path
          d="m480.683594 8h-35c13.152344 0 23.816406 10.664062 23.816406 23.816406v417.25c0 13.152344-10.664062 23.816406-23.816406 23.816406h35c13.152344 0 23.816406-10.664062 23.816406-23.816406v-417.25c0-13.152344-10.664062-23.816406-23.816406-23.816406zm0 0"
          fill="#e8e8e8"
        />
        <path
          d="m7.5 97.464844h497v-65.648438c0-13.152344-10.664062-23.816406-23.816406-23.816406h-449.367188c-13.152344 0-23.816406 10.664062-23.816406 23.816406zm0 0"
          fill="#2cab72"
        />
        <path
          d="m480.683594 8h-35c13.152344 0 23.816406 10.664062 23.816406 23.816406v65.648438h35v-65.648438c0-13.152344-10.664062-23.816406-23.816406-23.816406zm0 0"
          fill="#279463"
        />
        <path
          d="m84.808594 52.730469c0 9.75-7.902344 17.65625-17.652344 17.65625s-17.65625-7.90625-17.65625-17.65625 7.90625-17.652344 17.65625-17.652344 17.652344 7.902344 17.652344 17.652344zm0 0"
          fill="#fa5d72"
        />
        <path
          d="m144.570312 52.730469c0 9.75-7.90625 17.65625-17.65625 17.65625s-17.652343-7.90625-17.652343-17.65625 7.902343-17.652344 17.652343-17.652344 17.65625 7.902344 17.65625 17.652344zm0 0"
          fill="#f6d867"
        />
        <path
          d="m204.328125 52.730469c0 9.75-7.902344 17.65625-17.652344 17.65625s-17.65625-7.90625-17.65625-17.65625 7.90625-17.652344 17.65625-17.652344 17.652344 7.902344 17.652344 17.652344zm0 0"
          fill="#5994ce"
        />
        <path
          d="m402.8125 285.539062c0 81.082032-65.730469 146.8125-146.8125 146.8125s-146.8125-65.730468-146.8125-146.8125c0-81.082031 65.730469-146.816406 146.8125-146.816406s146.8125 65.734375 146.8125 146.816406zm0 0"
          fill="#495663"
        />
        <path
          d="m256 138.722656c-7.113281 0-14.101562.519532-20.945312 1.496094 71.164062 10.164062 125.871093 71.347656 125.871093 145.320312 0 73.96875-54.707031 135.15625-125.871093 145.320313 6.84375.976563 13.832031 1.492187 20.945312 1.492187 81.082031 0 146.8125-65.730468 146.8125-146.8125 0-81.082031-65.730469-146.816406-146.8125-146.816406zm0 0"
          fill="#3a444f"
        />
        <path
          d="m374.476562 285.539062c0 65.433594-53.042968 118.476563-118.476562 118.476563s-118.476562-53.042969-118.476562-118.476563c0-65.433593 53.042968-118.480468 118.476562-118.480468s118.476562 53.046875 118.476562 118.480468zm0 0"
          fill="#c0e4f0"
        />
        <path
          d="m256 167.058594c-4.796875 0-9.523438.292968-14.167969.847656 58.753907 7.003906 104.308594 56.996094 104.308594 117.632812 0 60.636719-45.554687 110.628907-104.308594 117.628907 4.644531.554687 9.371094.847656 14.167969.847656 65.433594 0 118.476562-53.042969 118.476562-118.476563 0-65.433593-53.042968-118.480468-118.476562-118.480468zm0 0"
          fill="#a6d6e2"
        />
        <path
          d="m256 211.625v-44.566406c-32.71875 0-62.335938 13.261718-83.777344 34.703125l31.511719 31.511719c13.378906-13.375 31.855469-21.648438 52.265625-21.648438zm0 0"
          fill="#f6d867"
        />
        <path
          d="m137.523438 285.539062h44.566406c0-20.410156 8.269531-38.886718 21.644531-52.261718l-31.511719-31.515625c-21.441406 21.441406-34.699218 51.058593-34.699218 83.777343zm0 0"
          fill="#2cab72"
        />
        <path
          d="m256 167.058594v44.566406c20.410156 0 38.886719 8.273438 52.261719 21.648438l31.515625-31.511719c-21.441406-21.441407-51.058594-34.703125-83.777344-34.703125zm0 0"
          fill="#fba826"
        />
        <path
          d="m324.40625 217.132812 15.371094-15.371093c-4.691406-4.691407-9.769532-8.988281-15.191406-12.84375-2.808594-1.996094-5.714844-3.867188-8.695313-5.621094-2.359375-1.382813-4.773437-2.6875-7.230469-3.910156-.886718-.441407-1.78125-.871094-2.683594-1.292969-.785156-.367188-1.578124-.722656-2.375-1.074219-1.011718-.445312-2.035156-.878906-3.066406-1.296875-1.257812-.507812-2.519531-1.003906-3.796875-1.472656-1.816406-.664062-3.652343-1.285156-5.507812-1.863281-2.78125-.863281-5.609375-1.632813-8.472657-2.292969-3.824218-.882812-7.714843-1.578125-11.667968-2.082031-4.941406-.628907-9.976563-.953125-15.089844-.953125v3.425781c28.058594 6.886719 52.191406 23.765625 68.40625 46.648437zm0 0"
          fill="#e09721"
        />
        <path
          d="m339.777344 201.761719-31.515625 31.511719c13.375 13.375 21.648437 31.855468 21.648437 52.265624h44.566406c0-32.71875-13.257812-62.335937-34.699218-83.777343zm0 0"
          fill="#fa5d72"
        />
        <path
          d="m324.40625 217.132812c13.6875 19.324219 21.734375 42.921876 21.734375 68.40625h28.335937c0-32.71875-13.257812-62.335937-34.699218-83.777343zm0 0"
          fill="#e35567"
        />
        <path
          d="m277.078125 285.539062c0 11.640626-9.4375 21.078126-21.078125 21.078126s-21.078125-9.4375-21.078125-21.078126c0-11.644531 9.4375-21.082031 21.078125-21.082031s21.078125 9.4375 21.078125 21.082031zm0 0"
          fill="#9c9a9b"
        />
        <path
          d="m92.308594 52.734375c0-13.871094-11.28125-25.15625-25.152344-25.15625s-25.15625 11.285156-25.15625 25.15625c0 13.867187 11.285156 25.152344 25.15625 25.152344 13.867188 0 25.152344-11.285157 25.152344-25.152344zm-25.152344 10.152344c-5.601562 0-10.15625-4.554688-10.15625-10.152344 0-5.601563 4.558594-10.15625 10.15625-10.15625s10.152344 4.554687 10.152344 10.15625c0 5.597656-4.554688 10.152344-10.152344 10.152344zm0 0"
        />
        <path
          d="m126.914062 77.886719c13.871094 0 25.15625-11.285157 25.15625-25.152344 0-13.871094-11.285156-25.15625-25.15625-25.15625-13.871093 0-25.152343 11.285156-25.152343 25.15625 0 13.867187 11.28125 25.152344 25.152343 25.152344zm0-35.308594c5.601563 0 10.15625 4.554687 10.15625 10.15625 0 5.597656-4.558593 10.152344-10.15625 10.152344-5.597656 0-10.152343-4.554688-10.152343-10.152344 0-5.601563 4.554687-10.15625 10.152343-10.15625zm0 0"
        />
        <path
          d="m186.675781 77.886719c13.867188 0 25.152344-11.285157 25.152344-25.152344 0-13.871094-11.285156-25.15625-25.152344-25.15625-13.871093 0-25.15625 11.285156-25.15625 25.15625 0 13.867187 11.285157 25.152344 25.15625 25.152344zm0-35.308594c5.597657 0 10.152344 4.554687 10.152344 10.15625 0 5.597656-4.554687 10.152344-10.152344 10.152344-5.601562 0-10.15625-4.554688-10.15625-10.152344 0-5.601563 4.554688-10.15625 10.15625-10.15625zm0 0"
        />
        <path
          d="m269.667969 60.234375h182.664062c4.144531 0 7.5-3.359375 7.5-7.5 0-4.144531-3.355469-7.5-7.5-7.5h-182.664062c-4.144531 0-7.5 3.355469-7.5 7.5 0 4.140625 3.355469 7.5 7.5 7.5zm0 0"
        />
        <path
          d="m480.683594.5h-449.367188c-17.265625 0-31.316406 14.046875-31.316406 31.316406v417.25c0 17.269532 14.050781 31.316406 31.316406 31.316406h449.367188c17.265625 0 31.316406-14.046874 31.316406-31.316406v-417.25c0-17.269531-14.050781-31.316406-31.316406-31.316406zm0 464.882812h-449.367188c-8.996094 0-16.316406-7.320312-16.316406-16.316406v-344.101562h41.808594c4.144531 0 7.5-3.359375 7.5-7.5 0-4.144532-3.355469-7.5-7.5-7.5h-41.808594v-58.148438c0-8.996094 7.320312-16.316406 16.316406-16.316406h449.367188c8.996094 0 16.316406 7.320312 16.316406 16.316406v58.148438h-412.191406c-4.140625 0-7.5 3.355468-7.5 7.5 0 4.140625 3.359375 7.5 7.5 7.5h412.191406v344.101562c0 8.996094-7.320312 16.316406-16.316406 16.316406zm0 0"
        />
        <path
          d="m381.976562 285.539062c0-33.382812-12.914062-65.097656-36.894531-89.082031-49.195312-49.195312-128.9375-49.222656-178.160156 0-23.605469 23.601563-36.898437 55.160157-36.898437 89.082031 0 69.464844 56.515624 125.976563 125.980468 125.976563 69.460938 0 125.972656-56.511719 125.972656-125.976563zm-80.527343-84.292968c-3.648438-1.964844-8.195313-.601563-10.164063 3.042968l-3.394531 6.296876c-7.699219-3.277344-15.910156-5.351563-24.390625-6.121094v-29.652344c24.964844 1.671875 47.695312 11.640625 65.453125 27.164062l-20.929687 20.929688c-2.230469-1.859375-4.546876-3.585938-6.9375-5.183594l3.402343-6.3125c1.96875-3.648437.605469-8.199218-3.039062-10.164062zm-52.949219 3.21875c-16.4375 1.492187-31.847656 7.875-44.523438 18.441406l-20.929687-20.929688c17.753906-15.523437 40.484375-25.492187 65.453125-27.167968zm-76.0625 8.117187 20.929688 20.933594c-10.566407 12.675781-16.949219 28.085937-18.441407 44.523437h-29.652343c1.675781-24.96875 11.640624-47.699218 27.164062-65.457031zm167.125 0c15.523438 17.757813 25.492188 40.488281 27.164062 65.457031h-29.652343c-1.492188-16.4375-7.875-31.847656-18.441407-44.523437zm-194.289062 80.457031h36.816406c4.140625 0 7.5-3.359374 7.5-7.5 0-17.738281 6.90625-34.417968 19.449218-46.960937.007813-.003906.011719-.007813.015626-.015625 12.542968-12.53125 29.214843-19.4375 46.945312-19.4375 8.609375 0 16.972656 1.628906 24.730469 4.742188l-18.242188 33.835937c-17.976562-4.1875-35.066406 9.542969-35.066406 27.835937 0 15.757813 12.820313 28.582032 28.578125 28.582032 25.734375 0 38.316406-31.558594 19.683594-49.285156l18.242187-33.832032c3.191407 2.226563 6.207031 4.753906 9.019531 7.5625.003907.003906.007813.007813.015626.015625 12.542968 12.542969 19.449218 29.21875 19.449218 46.957031 0 4.144532 3.359375 7.5 7.5 7.5h36.816406c-3.871093 57.710938-52.054687 103.480469-110.726562 103.480469-58.675781 0-106.855469-45.773437-110.726562-103.480469zm110.726562 6.078126c-7.488281 0-13.578125-6.089844-13.578125-13.578126 0-7.074218 5.433594-12.902343 12.347656-13.527343l-5.371093 9.964843c-1.964844 3.648438-.605469 8.195313 3.042968 10.164063 3.652344 1.96875 8.199219.59375 10.160156-3.042969l5.371094-9.964844c4.789063 8.921876-1.648437 19.984376-11.972656 19.984376zm0 0"
        />
        <path
          d="m256 131.222656c-85.089844 0-154.3125 69.226563-154.3125 154.316406 0 73.71875 52.449219 137.421876 124.714844 151.464844 4.066406.789063 8.003906-1.863281 8.792968-5.929687.789063-4.066407-1.867187-8.003907-5.933593-8.792969-65.230469-12.679688-112.574219-70.1875-112.574219-136.742188 0-76.820312 62.496094-139.316406 139.3125-139.316406s139.3125 62.496094 139.3125 139.316406c0 76.816407-62.496094 139.3125-139.234375 139.3125-.054687 0-.191406 0-.246094 0-4.144531 0-7.539062 3.355469-7.539062 7.5 0 4.140626 3.316406 7.5 7.460937 7.5h.246094c85.089844 0 154.3125-69.222656 154.3125-154.3125 0-85.089843-69.222656-154.316406-154.3125-154.316406zm0 0"
        />
      </svg>
    </div>
    <h1
      className="es7h93-1 eKtWtv"
    >
      Loading
    </h1>
  </div>,
  <div
    className="sc-1iqrqt3-0 gCMOPm"
  >
    <div
      data-test-id="iframe-mock"
    />
  </div>,
]
`;

exports[`EmbeddedIframeLayout should show an error block if loading has not completed within 10 seconds 1`] = `
<div
  className="es7h93-2 fWpopI"
>
  <div
    className="es7h93-0 fOAtDc"
  >
    <svg
      viewBox="1 -16 511.999 511"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m480.683594 472.882812h-449.367188c-13.152344 0-23.816406-10.664062-23.816406-23.816406v-417.25c0-13.15625 10.664062-23.816406 23.816406-23.816406h449.367188c13.15625 0 23.816406 10.660156 23.816406 23.816406v417.25c0 13.152344-10.664062 23.816406-23.816406 23.816406zm0 0"
        fill="#f4f4f4"
      />
      <path
        d="m480.683594 8h-35c13.152344 0 23.816406 10.660156 23.816406 23.816406v417.25c0 13.152344-10.664062 23.816406-23.816406 23.816406h35c13.152344 0 23.816406-10.664062 23.816406-23.816406v-417.25c0-13.15625-10.664062-23.816406-23.816406-23.816406zm0 0"
        fill="#e8e8e8"
      />
      <path
        d="m7.5 97.460938h497v-65.644532c0-13.15625-10.664062-23.816406-23.816406-23.816406h-449.367188c-13.152344 0-23.816406 10.660156-23.816406 23.816406zm0 0"
        fill="#fa5d72"
      />
      <path
        d="m480.683594 8h-35c13.152344 0 23.816406 10.660156 23.816406 23.816406v65.644532h35v-65.644532c0-13.15625-10.664062-23.816406-23.816406-23.816406zm0 0"
        fill="#e35567"
      />
      <path
        d="m84.808594 52.730469c0 9.75-7.902344 17.652343-17.652344 17.652343s-17.65625-7.902343-17.65625-17.652343 7.90625-17.652344 17.65625-17.652344 17.652344 7.902344 17.652344 17.652344zm0 0"
        fill="#2cab72"
      />
      <path
        d="m144.570312 52.730469c0 9.75-7.90625 17.652343-17.65625 17.652343s-17.652343-7.902343-17.652343-17.652343 7.902343-17.652344 17.652343-17.652344 17.65625 7.902344 17.65625 17.652344zm0 0"
        fill="#f6d867"
      />
      <path
        d="m204.328125 52.730469c0 9.75-7.902344 17.652343-17.652344 17.652343s-17.65625-7.902343-17.65625-17.652343 7.90625-17.652344 17.65625-17.652344 17.652344 7.902344 17.652344 17.652344zm0 0"
        fill="#5994ce"
      />
      <path
        d="m199.328125 264.441406c-9.753906 0-17.664063 7.90625-17.664063 17.664063v62c0 41.054687 33.28125 74.335937 74.335938 74.335937s74.335938-33.28125 74.335938-74.335937v-62c0-9.757813-7.910157-17.664063-17.664063-17.664063zm0 0"
        fill="#fa5d72"
      />
      <path
        d="m312.671875 264.441406h-31.570313c9.757813 0 17.664063 7.90625 17.664063 17.664063v62c0 35.636719-25.082031 65.398437-58.550781 72.640625 5.089844 1.097656 10.367187 1.695312 15.785156 1.695312 41.054688 0 74.335938-33.28125 74.335938-74.335937v-62c0-9.757813-7.910157-17.664063-17.664063-17.664063zm0 0"
        fill="#e35567"
      />
      <path
        d="m330.335938 246.777344c0-41.054688-33.28125-74.335938-74.335938-74.335938s-74.335938 33.28125-74.335938 74.335938c0 9.753906 7.910157 17.664062 17.664063 17.664062h113.34375c9.753906 0 17.664063-7.910156 17.664063-17.664062zm0 0"
        fill="#fa96a3"
      />
      <path
        d="m256 172.441406c-5.417969 0-10.695312.59375-15.785156 1.695313 33.46875 7.238281 58.550781 37.003906 58.550781 72.640625 0 9.753906-7.90625 17.664062-17.664063 17.664062h31.570313c9.757813 0 17.664063-7.910156 17.664063-17.664062 0-41.054688-33.28125-74.335938-74.335938-74.335938zm0 0"
        fill="#d17e89"
      />
      <path
        d="m92.308594 52.730469c0-13.871094-11.28125-25.152344-25.152344-25.152344s-25.15625 11.28125-25.15625 25.152344c0 13.871093 11.285156 25.152343 25.15625 25.152343 13.867188 0 25.152344-11.28125 25.152344-25.152343zm-25.152344 10.152343c-5.601562 0-10.15625-4.554687-10.15625-10.152343 0-5.597657 4.558594-10.152344 10.15625-10.152344s10.152344 4.554687 10.152344 10.152344c0 5.597656-4.554688 10.152343-10.152344 10.152343zm0 0"
      />
      <path
        d="m126.914062 77.882812c13.871094 0 25.15625-11.28125 25.15625-25.152343 0-13.871094-11.285156-25.152344-25.15625-25.152344-13.871093 0-25.152343 11.28125-25.152343 25.152344 0 13.871093 11.28125 25.152343 25.152343 25.152343zm0-35.304687c5.601563 0 10.15625 4.554687 10.15625 10.152344 0 5.597656-4.558593 10.152343-10.15625 10.152343-5.597656 0-10.152343-4.554687-10.152343-10.152343 0-5.597657 4.554687-10.152344 10.152343-10.152344zm0 0"
      />
      <path
        d="m186.675781 77.882812c13.867188 0 25.152344-11.28125 25.152344-25.152343 0-13.871094-11.285156-25.152344-25.152344-25.152344-13.871093 0-25.15625 11.28125-25.15625 25.152344 0 13.871093 11.285157 25.152343 25.15625 25.152343zm0-35.304687c5.597657 0 10.152344 4.554687 10.152344 10.152344 0 5.597656-4.554687 10.152343-10.152344 10.152343-5.601562 0-10.15625-4.554687-10.15625-10.152343 0-5.597657 4.554688-10.152344 10.15625-10.152344zm0 0"
      />
      <path
        d="m269.667969 60.230469h182.664062c4.144531 0 7.5-3.355469 7.5-7.5s-3.355469-7.5-7.5-7.5h-182.664062c-4.144531 0-7.5 3.355469-7.5 7.5s3.355469 7.5 7.5 7.5zm0 0"
      />
      <path
        d="m480.683594.5h-449.367188c-17.265625 0-31.316406 14.046875-31.316406 31.316406v417.25c0 17.265625 14.050781 31.316406 31.316406 31.316406h449.367188c17.265625 0 31.316406-14.050781 31.316406-31.316406v-417.25c0-17.269531-14.050781-31.316406-31.316406-31.316406zm0 464.882812h-449.367188c-8.996094 0-16.316406-7.320312-16.316406-16.316406v-344.105468h41.808594c4.144531 0 7.5-3.355469 7.5-7.5 0-4.140626-3.355469-7.5-7.5-7.5h-41.808594v-58.144532c0-8.996094 7.320312-16.316406 16.316406-16.316406h449.367188c8.996094 0 16.316406 7.320312 16.316406 16.316406v58.144532h-412.191406c-4.140625 0-7.5 3.359374-7.5 7.5 0 4.144531 3.359375 7.5 7.5 7.5h412.191406v344.105468c0 8.996094-7.320312 16.316406-16.316406 16.316406zm0 0"
      />
      <path
        d="m398.777344 317.542969c-1.40625-1.40625-3.3125-2.195313-5.304688-2.195313h-55.636718v-20.089844l51.253906-12.269531c2.972656-.710937 5.214844-3.15625 5.671875-6.179687l6.128906-40.804688c.613281-4.097656-2.207031-7.917968-6.304687-8.53125-4.089844-.621094-7.914063 2.203125-8.53125 6.300782l-5.375 35.804687-42.945313 10.28125c-.53125-5.984375-3.164063-11.371094-7.15625-15.417969 4.484375-4.546875 7.257813-10.785156 7.257813-17.664062 0-26.054688-12.238282-49.304688-31.269532-64.300782l26.925782-26.925781c2.929687-2.929687 2.929687-7.679687 0-10.605469-2.929688-2.929687-7.679688-2.929687-10.605469 0l-29.203125 29.199219c-23.511719-12.246093-51.6875-12.285156-75.25-.058593l-29.144532-29.140626c-2.929687-2.929687-7.675781-2.929687-10.605468 0-2.929688 2.929688-2.929688 7.675782 0 10.605469l26.847656 26.847657c-19.085938 14.996093-31.367188 38.28125-31.367188 64.378906 0 6.878906 2.773438 13.117187 7.261719 17.664062-3.992187 4.046875-6.628906 9.433594-7.160156 15.417969l-42.941406-10.28125-5.378907-35.800781c-.617187-4.097656-4.433593-6.914063-8.53125-6.304688-4.097656.613282-6.917968 4.433594-6.304687 8.53125l6.132813 40.804688c.453124 3.023437 2.695312 5.46875 5.667968 6.179687l51.253906 12.269531v20.089844h-55.636718c-1.988282 0-3.894532.792969-5.304688 2.199219l-27.890625 27.890625c-2.929687 2.929688-2.929687 7.679688 0 10.605469 2.929688 2.929687 7.679688 2.929687 10.605469 0l25.695312-25.695313h52.53125v13.757813c0 .988281.019532 1.972656.054688 2.953125l-51.308594 12.285156c-2.972656.710938-5.21875 3.15625-5.671875 6.179688l-6.128906 40.804687c-.683594 4.554687 2.847656 8.617187 7.425781 8.617187 3.648438 0 6.851563-2.667968 7.40625-6.386718l5.378906-35.804688 44.820313-10.730468c12.738281 56.8125 78.875 82.039062 126.136719 49.570312 3.414062-2.34375 4.28125-7.011719 1.9375-10.429688-2.347656-3.410156-7.011719-4.28125-10.429688-1.933593-43.753906 30.058593-104.621094-1.054688-104.621094-55.125v-54.644531c0-.089844 0-.179688 0-.269532v-7.085937c0-5.605469 4.558594-10.164063 10.164063-10.164063h113.34375c5.605469 0 10.164063 4.558594 10.164063 10.164063v62c0 13.328125-3.914063 26.203125-11.324219 37.226562-2.3125 3.441407-1.394531 8.097657 2.042969 10.410157s8.097656 1.394531 10.40625-2.042969c5.695312-8.476563 9.707031-17.839844 11.90625-27.675781l44.8125 10.730468 5.378906 35.800782c.613281 4.097656 4.429687 6.917968 8.53125 6.304687 4.09375-.617187 6.917968-4.4375 6.300781-8.53125l-6.128906-40.804687c-.453125-3.023438-2.699219-5.46875-5.671875-6.179688l-51.308594-12.285156c.089844-2.527344.039062-2.984375.054688-16.710938h52.53125l25.695312 25.695313c2.929688 2.929687 7.675781 2.929687 10.605469 0 2.929687-2.929688 2.929687-7.675781 0-10.605469zm-199.449219-60.601563c-5.605469 0-10.164063-4.5625-10.164063-10.164062 0-36.855469 29.980469-66.835938 66.835938-66.835938s66.835938 29.980469 66.835938 66.835938c0 5.601562-4.558594 10.164062-10.164063 10.164062zm0 0"
      />
    </svg>
  </div>
  <h1
    className="es7h93-1 eKtWtv"
  >
    Sorry, this tab didn't load in time
  </h1>
</div>
`;
